libres/tests/analysis/test_update.cpp

```diff
@@ -0,0 +1,173 @@
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <cmath>
+
+#include <catch2/catch.hpp>
+
+#include <ert/util/rng.h>
+#include <ert/enkf/enkf_util.hpp>
+
+#include <ert/analysis/update.hpp>
+
+namespace analysis {
+void run_analysis_update_without_rowscaling(
+    analysis_module_type *module, const bool_vector_type *ens_mask,
+    const meas_data_type *forecast, obs_data_type *obs_data,
+    rng_type *shared_rng, matrix_type *E,
+    std::unordered_map<std::string, matrix_type *> parameters);
+}
+
+const double a_true = 5.0;
+const double b_true = 0.5;
+
+struct model {
+    double a;
+    double b;
+
+    model(double a, double b) : a(a), b(b) {}
+
+    model(rng_type *rng) {
+        this->a = enkf_util_rand_normal(a_true, 10.0, rng);
+        this->b = enkf_util_rand_normal(b_true, 10.0, rng);
+    }
+
+    double eval(double x) const { return this->a * x + this->b; }
+};
+
+#include <iostream>
+#include <fmt/format.h>
+
+SCENARIO("Running analysis update without row scaling on linear model",
+         "[analysis]") {
+    GIVEN("Increasing ensemble size") {
+        int ens_size = GENERATE(100, 1000);
```

Comment:
When you systematically vary both obs_std and ens_size I would split the two clearer in the code - e.g. something like:
```
int ens_size = GENERATE(100,1000);
// All setup which is related to the ensemble
THEN("very observation std") {
     double obs_std = GENERATE(0.000001, 0.5, 10000);
     // the rest
}
```

Reply:
Not sure how I can create this split as `obs_std` is used to in the creation of observations, which are needed for every test.
This is why I put all the `GENERATE` under the `GIVEN`-keyword.
Does that make sense?